### ISSUE TITLE
application of RAFT-stereo for selfsupervised learning

### ISSUE THREAD

**Issue by HaiLiExp** [princeton-vl/RAFT-Stereo#85]
Hello

I work on self-supervised learning for depth estimation. The only difference to supervised learning is the more complicated calculation of the loss function: instead of a comparison with ground truth, I do backprojection of the image pixels into 3D space and projection again into the other camera. I tried mobilestereonet, it works fine, but RAFT-stereo learns nothing if the training begins from scratch.

Not sure whether you have read my former text. I found out that the finetuning should have a much lower learning rate than you proposed. I can get qualitatively reasonable results. The question now is how to make the training from scratch.

One difference to RAFT-stereo is that the training loss is not a summation, but an average of the "error map". Is that a problem?

**Comment by HaiLiExp** [comment]
Hello 

I started with the pretrained model with "--restore_ckpt models/raftstereo-middlebury.pth". It is very strange that the loss becomes greater and the predictions become worse after 1000 and 2000 steps with batch size of 2.
![flow_preds1 0 2000](https://user-images.githubusercontent.com/71226859/228807429-ded80f03-dad7-41a3-991e-dc91717f7a02.jpg)
![flow_preds1 0 1000](https://user-images.githubusercontent.com/71226859/228807433-f733f080-2906-44fc-8cf5-67cb7cffc2c9.jpg)
![flow_preds1 0 0](https://user-images.githubusercontent.com/71226859/228807437-b7f27e5a-49e0-4d33-9688-25e93ef1c01a.jpg)

**Comment by HaiLiExp** [comment]
Compared to RAFT-stereo, the only thing I modify is the calculation of the loss function, instead of a comparison with the ground truth, I calculate the reconstruction loss without trainable parameters. I don't understand why the model performace becomes worse as training continues.
